src/lib/updating-element.ts

```diff
@@ -157,8 +157,19 @@ export const defaultConverter: ComplexAttributeConverter = {
       case Number:
         return value === null ? null : Number(value);
       case Object:
+        try {
+          return JSON.parse(value!);
+        } catch (e) {
+          // Ignore any previous errors. Only return a empty object
+          return {};
```

Comment:
I don't think we should be returning empty objects/arrays in this case. Downstream code would not be able to differentiate between an error value and an empty object. I think `undefined` is a better fallback, even though it's still valid JSON.

Another option is to add an `onError` property to `PropertyDeclaration` and call it here.

Reply:
onError property sounds good